src/VisualStudio/Core/Def/Implementation/AbstractVsTextViewFilter.cs

```diff
@@ -119,7 +119,10 @@ protected int GetDataTipTextImpl(ITextBuffer subjectBuffer, TextSpan[] pSpan, ou
                             var spanOpt = textSnapshot.TryGetSpan(textSpan);
                             if (spanOpt.HasValue)
                             {
-                                var dataTipInfo = languageDebugInfo.GetDataTipInfoAsync(document, spanOpt.Value.Start, cancellationToken).WaitAndGetResult(cancellationToken);
+                                // 'kind' is an lsp-only concept, so we don't want/need to include it here (especially
+                                // as it can be expensive to compute, and we don't want to block the UI thread).
+                                var dataTipInfo = languageDebugInfo.GetDataTipInfoAsync(
```

Comment:
if we're returning data tips from LSP, why are we computing it again here in a MEFish API?   Probably missing something about the design here

Reply:
asnwered offline.  will support the old and new systems for a short period of time.